### PR TITLE
Add overview to docs

### LINKED ISSUE #690: qiskit-nature documentation: overview page
The current "landing page" for qiskit-nature is a TOC that does not include any overview about the package. 

Screenshot 2022-06-16 at 15 12 43

Compare it with [qiskit-ibm-runtime](https://qiskit.org/documentation/partners/qiskit_ibm_runtime/):

Screenshot 2022-06-16 at 15 10 39

Qiskit nature docs needs an overview page, explaining the goal of the package, as well as the key concepts to newcomers.

### PR DESCRIPTION
**Pull request by manoelmarques**
### Summary
Fixes #690

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -1,14 +1,38 @@
...
-###########################
+######################
+Qiskit Nature overview
+######################
+
+Overview
+==============
+
+**Qiskit Nature** is an open-source framework that supports problems including ground state energy computations,
+excited states and dipole moments of molecule, both open and closed-shell.
```
Comment (mrossinek) [inline docs/index.rst:9]:
```suggestion
**Qiskit Nature** is an open-source framework which supports solving natural science problems
using quantum computing algorithms. This includes finding ground and excited states of electronic
and vibrational structure problems, measuring the dipole moments of molecular systems, solving
the Ising and Fermi-Hubbard models on lattices, and much more.
```

File: docs/index.rst [hunk 2]
```diff
@@ -1,14 +1,38 @@
...
+
+The code comprises chemistry drivers, which when provided with a molecular
+configuration will return one and two-body integrals as well as other data that is efficiently
```
Comment (mrossinek) [inline docs/index.rst:12]:
```suggestion
configuration will return one- and two-body integrals as well as other data that is efficiently
```

File: docs/index.rst [hunk 3]
```diff
@@ -1,14 +1,38 @@
...
+computed classically. This output data from a driver can then be used as input in Qiskit
+Nature that contains logic which is able to translate this into a form that is suitable
```
Comment (mrossinek) [inline docs/index.rst:14]: I'm not yet too happy with the phrasing of this sentence but also don't have an immediate suggestion on how to improve it.. :thinking:

File: docs/index.rst [hunk 4]
```diff
@@ -1,14 +1,38 @@
...
+for quantum algorithms. The conversion first creates a FermionicOperator which must then be mapped,
+e.g. by a Jordan Wigner mapping, to a qubit operator in readiness for the quantum computation.
```
Comment (mrossinek) [inline docs/index.rst:16]:
```suggestion
for quantum algorithms. The conversion first creates a fermionic operator which must then be mapped,
e.g. by a Jordan Wigner mapping, to a qubit operator in readiness for the quantum computation.
```

This aligns better with the qubit operator mentioned later. If we don't want this, we should say `FermionicOp`, matching the current name of the class.

**Comment by mrossinek** [comment]
I'm wondering if it might make sense to hold off on this, until the major refactoring to be done for 0.5 :thinking:

**Comment by manoelmarques** [comment]
> I'm wondering if it might make sense to hold off on this, until the major refactoring to be done for 0.5 🤔

Since the other apps are changing to this format too, we could do it now, upload to docs and then update it after refactoring.

**Comment by woodsp-ibm** [comment]
Maybe there is more general material from this article that can be used here https://medium.com/qiskit/introducing-qiskit-nature-cb9e588bb004

File: README.md [hunk 5]
```diff
@@ -2,14 +2,16 @@
 
 [![License](https://img.shields.io/github/license/Qiskit/qiskit-nature.svg?style=popout-square)](https://opensource.org/licenses/Apache-2.0)<!--- long-description-skip-begin -->[![Build Status](https://github.com/Qiskit/qiskit-nature/workflows/Nature%20Unit%20Tests/badge.svg?branch=main)](https://github.com/Qiskit/qiskit-nature/actions?query=workflow%3A"Nature%20Unit%20Tests"+branch%3Amain+event%3Apush)[![](https://img.shields.io/github/release/Qiskit/qiskit-nature.svg?style=popout-square)](https://github.com/Qiskit/qiskit-nature/releases)[![](https://img.shields.io/pypi/dm/qiskit-nature.svg?style=popout-square)](https://pypi.org/project/qiskit-nature/)[![Coverage Status](https://coveralls.io/repos/github/Qiskit/qiskit-nature/badge.svg?branch=main)](https://coveralls.io/github/Qiskit/qiskit-nature?branch=main)<!--- long-description-skip-end -->
 
-**Qiskit Nature** is an open-source framework that supports problems including ground state energy computations,
-excited states and dipole moments of molecule, both open and closed-shell.
+**Qiskit Nature** is an open-source framework which supports solving natural science problems
```
Comment (mrossinek) [inline README.md:5]:
```suggestion
**Qiskit Nature** is an open-source framework which supports solving quantum mechanical natural science problems
```

File: docs/index.rst [hunk 6]
```diff
@@ -1,14 +1,40 @@
...
-Qiskit Nature Documentation
-###########################
+######################
+Qiskit Nature overview
+######################
+
+Overview
+==============
+
+**Qiskit Nature** is an open-source framework which supports solving natural science problems
```
Comment (mrossinek) [inline docs/index.rst:8]:
```suggestion
**Qiskit Nature** is an open-source framework which supports solving quantum mechanical natural science problems
```

File: docs/index.rst (hunk 3 above)
Comment (mrossinek) [reply docs/index.rst:16]: How about something like the following:
> Qiskit Nature provides logic to interpret this output data, translating it into a form that is suitable for quantum algorithms.

File: docs/index.rst [hunk 7]
```diff
@@ -1,14 +1,40 @@
...
+using quantum computing algorithms. This includes finding ground and excited states of electronic
+and vibrational structure problems, measuring the dipole moments of molecular systems, solving
+the Ising and Fermi-Hubbard models on lattices, and much more.
+
+The code comprises chemistry drivers, which when provided with a molecular
+configuration will return one and two-body integrals as well as other data that is efficiently
+computed classically. This output data from a driver can then be used as input in Qiskit
+Nature that contains logic which is able to translate this into a form that is suitable
+for quantum algorithms. The conversion first creates a fermionic operator which must then be mapped,
+e.g. by a Jordan Wigner mapping, to a qubit operator in readiness for the quantum computation.
```
Comment (mrossinek) [inline docs/index.rst:18]: I would rephrase this similar to the following:
> For the solution of electronic structure problems, the problem Hamiltonian is first expressed in the second quantization formalism, comprising fermionic excitation and annihilation operators. These can then be mapped to the qubit formalism using a variety of mappings such as Jordan-Wigner, Parity, and more, in readiness for the quantum computation.

File: docs/index.rst [hunk 8]
```diff
@@ -1,14 +1,40 @@
...
+
```
Comment (mrossinek) [inline docs/index.rst:19]: Now we should add a section on vibrational structure calculations, which can also result from the data obtained by a classical driver.

And finally, we need to mention the lattice problems, which do not derive from classically obtained data.

**Review by mrossinek — APPROVED** [review]
Okay, what is here is good enough for now :+1: 

We can improve upon this further but let's get this overview published as is now and tackle this during the documentation upgrading after the ongoing refactorings :+1:
